paddle/inference/example.cc

```diff
@@ -28,12 +30,29 @@ int main(int argc, char** argv) {
     exit(1);
   }
 
+  // 1. Define place, executor, scope
+  auto* place = new paddle::platform::CPUPlace();
```

Comment:
Thanks for the review. Quick question: why do we want to use `auto` over `auto *` or vice-versa?

Reply:
I think the question is more about when do you want to create a instance of a class or new a instance of class. `auto` and `auto*` will follow naturally. Below is just my personal opinion. 

For this place example, because place is only used as a variable to separate CPU/GPU, we don't need to modify its state or use its member function. To simplify the code, we prefer to directly construct `paddle::platform::CPUPlace()`.

For other examples, where you may want to modify the state of a instance or use its member functions, it may be better to use `auto*  = new xxx`.

Reply:
Done